content/_index.md

```diff
@@ -419,9 +419,14 @@ into the `m.` namespace.
 
 ### Timestamps
 
-Unless otherwise stated, timestamps are measured as milliseconds since
-the Unix epoch. Throughout the specification this may be referred to as
-POSIX, Unix, or just "time in milliseconds".
+Unless otherwise stated, timestamps are [Unix
+timestamps](https://en.wikipedia.org/wiki/Unix_time), but measured in
+milliseconds. This means, they approximate the number of milliseconds
+since 1970-01-01 00:00:00.000 UTC, but disregard leap seconds so that
+each day is precisely 86,400,000 milliseconds. This also means that
+timestamps can repeat. Most programming languages provide timestamps in
+that format natively. Throughout the specification this may be referred
+to as POSIX, Unix, or just "time in milliseconds".
```

Comment:
Wikipedia seems to be fine with calling them Unix timestamps:
> Unix time[[a]](https://en.wikipedia.org/wiki/Unix_time#cite_note-5) is a date and time representation widely used in [computing](https://en.wikipedia.org/wiki/Computing). It measures time by the number of [seconds](https://en.wikipedia.org/wiki/Second) that have elapsed since 00:00:00 [UTC](https://en.wikipedia.org/wiki/Coordinated_Universal_Time) on 1 January 1970, the Unix [epoch](https://en.wikipedia.org/wiki/Epoch_(computing)), without adjustments made due to [leap seconds](https://en.wikipedia.org/wiki/Leap_second). In modern computing, values are **sometimes stored with higher [granularity](https://en.wikipedia.org/wiki/Data_granularity), such as [microseconds](https://en.wikipedia.org/wiki/Microseconds) or [nanoseconds](https://en.wikipedia.org/wiki/Nanoseconds)**.

> Can we just phrase it as "timestamps are the number of milliseconds elapsed since the unix epoch (1970-01-01 00:00:00 UTC), disregarding leap seconds." ?

I think the way timestamps are defined is very subtle wrt. leap seconds. What does "disregarding leap seconds" mean? Just ignoring them entirely? No, you actively have to know whether you're in a leap second to repeat that second. Of course, since the normal time-related functions on the OSs do that for you, that's easy for the programmer. But I think your sentence is a bit too unclear. The timestamp is absolutely not the number of milliseconds elapsed since the unix epoch, the "disregarding leap seconds" is doing a lot of work there.

Reply:
> Unix time[[a]](https://en.wikipedia.org/wiki/Unix_time#cite_note-5) is a date and time representation widely used in [computing](https://en.wikipedia.org/wiki/Computing). It measures time by the number of [seconds](https://en.wikipedia.org/wiki/Second) that have elapsed since 00:00:00 [UTC](https://en.wikipedia.org/wiki/Coordinated_Universal_Time) on 1 January 1970, the Unix [epoch](https://en.wikipedia.org/wiki/Epoch_(computing)), without adjustments made due to [leap seconds](https://en.wikipedia.org/wiki/Leap_second). In modern computing, values are sometimes stored with higher [granularity](https://en.wikipedia.org/wiki/Data_granularity), such as [microseconds](https://en.wikipedia.org/wiki/Microseconds) or [nanoseconds](https://en.wikipedia.org/wiki/Nanoseconds).

... but that doesn't say that unix timestamps are measured in microseconds or nanoseconds; rather, it says that timestamps may be represented in some *other format* that has higher granularity. I still don't think it's correct to say "unix timestamps but in milliseconds".

> What does "disregarding leap seconds" mean? Just ignoring them entirely? No, you actively have to know whether you're in a leap second to repeat that second.

Well, that's up to you. 

I think it's fairly clear what "the number of milliseconds elapsed since the unix epoch ... disregarding leap seconds" means for all times that are *not* in the middle of a leap second, and I've also said that what happens *during* a leap second should be left for implementations to decide.

It's interesting to note that the ECMAScript (ie, javascript) spec has a [whole section](https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-time-values-and-time-range) devoted to the subtleties of this ... which makes me wonder if the index should say something short, with a link to a section in the appendices where we can spell out exactly what we mean.

Reply:
> I think it's fairly clear what "the number of milliseconds elapsed since the unix epoch ... disregarding leap seconds" means for all times that are _not_ in the middle of a leap second

I don't think the sentence makes it clear. Maybe we could call it "not counting leap seconds"?

I like the idea of linking to the ECMAScript spec. Is there some kind of link that looks permanent enough to include it in the spec? Is that link already permanent enough?

Reply:
> I don't think the sentence makes it clear. Maybe we could call it "not counting leap seconds"?

sure.

> I like the idea of linking to the ECMAScript spec. Is there some kind of link that looks permanent enough to include it in the spec? Is that link already permanent enough?

Yes I think that link is permanent "enough", but I wouldn't be particularly keen on using the ECMAScript spec as the primary definition. Happy for us to give our own definition and then say "this is the same as the definition in ECMAScript", though.

All I really meant, by reference to the ECMAScript spec, is to note that they give a lot of detail. And I think that, if we wanted to give that much detail, I'd put it in the appendices rather than the index.